src/distributions/range.rs

```diff
@@ -99,7 +98,7 @@ macro_rules! integer_impl {
 
             fn construct_range(low: $ty, high: $ty) -> Range<$ty> {
                 let range = (w(high as $unsigned) - w(low as $unsigned)).0;
-                let unsigned_max: $unsigned = Int::max_value();
+                let unsigned_max: $unsigned = -1 as $unsigned;
```

Comment:
This can probably leverage the `MAX` constants (e.g. http://doc.rust-lang.org/std/i16/constant.MAX.html).